scribejava-apis/src/main/java/com/github/scribejava/apis/service/ProductHuntServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package com.github.scribejava.apis.service;
+
+import com.github.scribejava.core.builder.api.DefaultApi20;
+import com.github.scribejava.core.model.OAuthConfig;
+import com.github.scribejava.core.oauth.OAuth20ServiceImpl;
+
+public class ProductHuntServiceImpl extends OAuth20ServiceImpl {
+
+    public ProductHuntServiceImpl(final DefaultApi20 api, final OAuthConfig config) {
+        super(api, config);
+    }
+    
+    @Override
+    protected <T extends AbstractRequest> T createAccessTokenRequest(final Verifier verifier, T request) {
+        super.createAccessTokenRequest(verifier, request);
+        if (!getConfig().hasGrantType()) {
+            request.addParameter(OAuthConstants.GRANT_TYPE, "authorization_code");
```

Comment:
@kullfar Could you please send me the email address which can be used to send you the credential? Thanks